Automattic/jetpack
# WordPress.com Toolbar: Add Site Logs menu gated behind a constant

Fixes Automattic/dotcom-forge#1987

## Proposed changes:

* Jetpack plugin can add an optional new Site Logs menu item to the Tools menu.
* Site Logs is only ever added on WoA sites
* Menu item is hidden by default and can be added with the `jetpack_show_wpcom_site_logs_menu` filter

<img width="332" alt="CleanShot 2023-03-22 at 17 57 00@2x" src="https://user-images.githubusercontent.com/1500769/226806156-40bfd11c-5af2-49bf-b2f8-a477f2f80a57.png">

The filter is handled in Automattic/wpcomsh#1315

Having the logic in `wpcomsh` means we can quickly show the menu item to all users without waiting for a Jetpack release.

### Other information:


## Jetpack product discussion


p9dueE-6YC-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






When testing be sure to check the menu both in Calypso and wp-admin.

* [Run these commands on your sandbox](#issuecomment-1478892176) and confirm the Tools menu is unchanged for a sandboxed Simple sites.
* Test on a non-WoA Jetpack site with JN. You can create one with this branch's build already installed. ![CleanShot 2023-03-22 at 17 32 46@2x](https://user-images.githubusercontent.com/1500769/226803416-c6227b05-faa6-424b-811f-0a6f993af5a2.png)
* Test WoA using a dev blog (other docs about testing this on WoA PCYsg-Osp-p2)
  * Install [Jetpack Beta Tester Plugin](https://jetpack.com/download-jetpack-beta/)
  * Configure beta plugin `/wp-admin/admin.php?page=jetpack-beta`
  * Configure Jetpack to use the `add/site-logs-menu` feature branch
  * Deploy Automattic/wpcomsh#1315 to your WoA dev blog
  * If you are on the dev team specified in `wpcomsh`, you should now see the Site Logs menu item in Calypso and wp-admin
  * (after clicking the Site Logs menu item you might be redirected to `/home` unless your testing in a version of Calypso where the page is enabled)
